base/sparse/sparsematrix.jl

```diff
@@ -2597,10 +2600,12 @@ function setindex!{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, x, I::AbstractMatrix{Bool})
 
                 if (mode > 1) && (nadd == 0) && (ndel == 0)
                     # copy storage to take changes
-                    colptrB = copy(colptrA)
+                    colptrA = copy(colptrB)
                     memreq = (x == 0) ? 0 : n
-                    rowvalB = Array(Ti, length(rowvalA)+memreq); copy!(rowvalB, 1, rowvalA, 1, r1-1)
-                    nzvalB = Array(Tv, length(nzvalA)+memreq); copy!(nzvalB, 1, nzvalA, 1, r1-1)
+                    rowvalA = copy(rowvalB)
+                    nzvalA = copy(nzvalB)
+                    resize!(rowvalB, length(rowvalA)+memreq)
+                    resize!(nzvalB, length(rowvalA)+memreq)
```

Comment:
Do I understand correctly that in this method `colptrB`, `rowvalB`, and `nzvalB` are now persistent references to `A.colptr`, `A.rowval`, and `A.nzval`, while `colptrA`, `rowvalA`, and `nzvalA` can become disjoint storage for the original contents of `A.colptr`, `A.rowval`, and `A.nzval`? If so, this being a bit labyrinthine, I might advocate for renaming things commensurate with their use as in the preceding methods.